### ISSUE TITLE
[bug-website module] Comick some chapters fail to download

### ISSUE THREAD

**Issue by Kummo33** [dazedcat19/FMD#1380]
### Prerequisites

- [X] I'm using the latest version.
- [X] All website modules are up to date.
- [X] This is not temporary server issue.

### System information

Windows 7 x64

### Describe the bug

Some chapters of manga can't be downloaded in FMD. Can be viewed normally on the website though.

Eg https://comick.cc/comic/00-helck
Can download Ch. 041 [Official] and all chapters prior, but bugs out at Ch. 042 [Official]- Ch. 052.5 [Official]

Eg https://comick.cc/comic/00-ultimate-scheming-system
Can download chs up to c625 LQ, but bugs out at c626 LQ- c634 LQ

Also does anyone know how to get FMD to output a log when encountering these download errors like in other open issues? It would be of great help in troubleshooting further.

**Comment by Slasar41** [comment]
The API provide .webp images but the browser got .jpg images.
![image](https://github.com/dazedcat19/FMD2/assets/28888646/14dc44a6-516f-4a4c-a0ee-a3b2b5e8aca3)
![image](https://github.com/dazedcat19/FMD2/assets/28888646/635c5721-c702-4be9-81f7-d7c24f4918a5)
All the working chapters from the examples also provide .webp but it's working so I assume this is a CDN issue.

**Comment by Kummo33** [comment]
How strange. Any idea on how to solve this? Like maybe some way to have FMD pull the jpeg files like in the browser version?

Here's what I get if I try to direct link the said image in browser:
![error](https://github.com/dazedcat19/FMD2/assets/53865835/a26c84ab-315c-4714-94cf-a20d4b8122b8)

![error2](https://github.com/dazedcat19/FMD2/assets/53865835/4edfcbf5-5c6f-453c-ba44-5e81aca47f43)

Here's the link to jpeg version which can be viewed correctly:
![error3](https://github.com/dazedcat19/FMD2/assets/53865835/094e16de-0af0-48a3-9680-577a39801593)

**Comment by Kummo33** [comment]
Oops wrong button accidentally closed issue.

**Comment by Slasar41** [comment]
I don't know how the browser got .jpg files, the API that FMD uses provide .webp files which curently doesn't exist on CDN server.
Unless there's a condition to switch to .jpg files from the API, which I don't see it.
Here's the API response for Helck - Ch. 041 [Official]: https://api.comick.ink/chapter/E_DZR05L
Here's the API response for Helck - Ch. 042 [Official]: https://api.comick.ink/chapter/k9PZZXwC
You can force FMD to be able to download the broken chapter by changing the line 131 to `TASK.PageLinks.Add(CDN_URL .. '/' .. v.ToString():gsub('.webp', '-m.jpg'))` in `lua\modules\ComicK.lua`. But this will make the normal chapter broken, so only change it to download the broken chapter.

**Comment by Kummo33** [comment]
Nice! The missing chapters downloaded properly after changing the line. 

Thank you for the workaround.

**Comment by enchained** [comment]
> You can force FMD to be able to download the broken chapter by changing the line 131 to `TASK.PageLinks.Add(CDN_URL .. '/' .. v.ToString():gsub('.webp', '-m.jpg'))` in `lua\modules\ComicK.lua`. But this will make the normal chapter broken, so only change it to download the broken chapter.

Thank you for the info. I encountered the issue recently on multiple chapters of different titles, so using the workaround does not feel very practical. I noticed that API docs here mentioned tachiyomi=true key https://api.comick.fun/docs/static/index.html

Looks like correct URLs (CDN included) are returned when using this key:
https://api.comick.fun/chapter/E_DZR05L?tachiyomi=true
https://api.comick.fun/chapter/k9PZZXwC?tachiyomi=true
But using this key will require to modify the existing data structure.

Is there a reason this key is not being used?
